packages/backend/src/core/preprocessing/SpotHistoryPreprocessor.ts

```diff
@@ -32,13 +32,19 @@ export class SpotHistoryPreprocessor extends HistoryPreprocessor<AssetHash> {
       if (vault.starkKey === StarkKey.ZERO) {
         await this.closePositionOrVault(trx, vault.vaultId, stateUpdate, {})
       } else {
-        const currentRecord =
-          await this.preprocessedAssetHistoryRepository.findCurrentByStarkKeyAndAsset(
-            vault.starkKey,
-            vault.assetHash,
+        const currentRecords =
+          await this.preprocessedAssetHistoryRepository.getCurrentByPositionOrVaultId(
+            vault.vaultId,
```

Comment:
Do we want to change find to get?

Reply:
We need to, find* returns one record, get* returns a list.

Reply:
Right, we now have a list, my bad